lib/jekyll/commands/new.rb

```diff
@@ -21,6 +22,12 @@ def init_with_program(prog)
         def process(args, options = {})
           raise ArgumentError, "You must specify a path." if args.empty?
 
+          if args.size == 1
+            new_blog_title = args[0]
```

Comment:
There is nothing stopping you from having an uppercase dir name anyways.
`the_blue_horn`, `The Blue Horn`, `THE BLUE HORN` is all supported. 
Pass what you like as arguments.